preprocessors/ocr/ocr.py

```diff
@@ -0,0 +1,152 @@
+from azure.cognitiveservices.vision.computervision import ComputerVisionClient
+from azure.cognitiveservices.vision.computervision.models import OperationStatusCodes
+from azure.cognitiveservices.vision.computervision.models import VisualFeatureTypes
+from msrest.authentication import CognitiveServicesCredentials
+
+app = Flask(__name__)
+
+
+@app.route('/preprocessor', methods=['POST', 'GET'])
+def get_ocr_text():
+    """
+    Gets data on locations nearby a map from the Autour API
+    """
+    # Load schemas
+    with open('./schemas/preprocessors/ocr.schema.json') as jsonfile:
+        data_schema = json.load(jsonfile)
+    with open('./schemas/preprocessor-response.schema.json') as jsonfile:
+        schema = json.load(jsonfile)
+    with open('./schemas/definitions.json') as jsonfile:
+        definition_schema = json.load(jsonfile)
+    schema_store = {
+        data_schema['$id']: data_schema,
+        schema['$id']: schema,
+        definition_schema['$id']: definition_schema
+    }
+    content = request.get_json()
+    with open('./schemas/request.schema.json') as jsonfile:
+        request_schema = json.load(jsonfile)
+    # Validate incoming request
+    resolver = jsonschema.RefResolver.from_schema(
+            request_schema, store=schema_store)
+    try:
+        validator = jsonschema.Draft7Validator(
+            request_schema,
+            resolver=resolver
+        )
+        validator.validate(content)
+    except jsonschema.exceptions.ValidationError as error:
+        logging.error(error)
+        return jsonify("Invalid Request JSON format"), 400
+    # Use response schema to validate response
+    resolver = jsonschema.RefResolver.from_schema(
+            schema, store=schema_store)
+    # Get OCR text response
+    ocr_result = get_ocr_text(content['image'])
+    
+    if ocr_result is None:
+        return jsonify("Could not retreive Azure results"), 400
+
+    name = 'ca.mcgill.a11y.image.preprocessor.ocr'
+    request_uuid = content['request_uuid']
+    timestamp = int(time.time())
+    data = {'lines': ocr_result}
+
+    try:
+        validator = jsonschema.Draft7Validator(data_schema, resolver=resolver)
+        validator.validate(data)
+    except jsonschema.exceptions.ValidationError as error:
+        logging.error(error)
+        return jsonify("Invalid Preprocessor JSON format"), 500
+
+    response = {
+        'request_uuid': request_uuid,
+        'timestamp': timestamp,
+        'name': name,
+        'data': data
+    }
+
+    try:
+        validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+        validator.validate(response)
+    except jsonschema.exceptions.ValidationError as error:
+        logging.error(error)
+        return jsonify("Invalid Preprocessor JSON format"), 500
+
+    return response
+
+def get_ocr_text(source):
+    """
+    Gets OCR text data from Azure API
+    """
+
+    # Convert URI to binary stream
+
+    image_b64 = source.split(",")[1]
```

Comment:
See comment on line 124.

Reply:
This is fine. Request schema ensures that this will work